app/src/main/java/net/gsantner/markor/format/markdown/MarkdownReplacePatternGenerator.java

```diff
@@ -30,7 +31,13 @@ public class MarkdownReplacePatternGenerator {
     public static final Pattern PREFIX_UNORDERED_LIST = Pattern.compile("^(\\s*)((-|\\*|\\+)\\s)");
     public static final Pattern PREFIX_LEADING_SPACE = Pattern.compile("^(\\s*)");
 
-    static final Pattern[] PREFIX_PATTERNS = {
+    public static final AutoTextFormatter.FormatPatterns formatPatterns = new AutoTextFormatter.FormatPatterns(
```

Comment:
Dramatically simplify the number of files / classes / indirection used for the patterns. We can simplify them further in the future, but this is a good stopping point.